cellpack/autopack/loaders/migrate_v1_to_v2.py

```diff
@@ -67,7 +67,7 @@ def migrate_ingredient(old_ingredient):
     for attribute in list(old_ingredient):
         if attribute in v1_to_v2_name_map:
             if attribute == "Type":
-                value = ingredient_types_map[old_ingredient[attribute]]
+                value = ingredient_types_map[old_ingredient[attribute]].value
```

Comment:
why did you make this change?

Reply:
Before making this change, I tried to call `_save_converted_recipe` in `_migrate_version`, it raised TypeError: Object of type INGREDIENT_TYPE is not JSON serializable. So we add `.value` to get the value str in group programming. Please let me know if there's better ways to do this, I can look into it.

Reply:
yeah, so when you save it, you can convert it to a string. But you shouldn't do that to all of the code, just in `_save_converted_recipe`

Reply:
v2 recipes do not go through `migrate_ingredient` and have the ingredient type as string. This can cause issues while creating ingredients in the Environment (comparing enum to strings). We should think about converting everything to enum if we want to avoid using string representations. Thoughts?

Reply:
Yeah, we can convert them on load, which can be part of the check to make sure they're an allowed member. I'll make an issue.

Reply:
Although this check works: `INGREDIENT_TYPE.SINGLE_SPHERE == "single_sphere"`

Reply:
This is now fixed in a new PR